src/resources/asres.rs

```diff
@@ -306,14 +337,54 @@ impl FromStr for AsBlocks {
 
         for el in s.split(',') {
             let el = el.trim();
-            let block = AsBlock::from_str(&el)?;
-            builder.push(block);
+            if !el.is_empty() {
+                let block = AsBlock::from_str(&el)?;
+                builder.push(block);
+            }
         }
 
         Ok(builder.finalize())
     }
 }
 
+//--- PartialEq
+
+impl PartialEq for AsBlocks {
+    fn eq(&self, other: &AsBlocks) -> bool {
+        // Relying on the fact that blocks are kept in order
+        let mut other_iter = other.iter();
+        for my_block in self.iter() {
+            if let Some(other_block) = other_iter.next() {
+                if my_block.min() != other_block.min() || my_block.max() != other_block.max() {
+                    return false
+                }
+            } else {
+                return false
+            }
+        }
+        true
+    }
+}
+
+impl Eq for AsBlocks {}
+
+//--- Serialize
+
+impl Serialize for AsBlocks {
```

Comment:
Unless you totally want them to be serialized in their string representation, you can probably derive {Des,S}erialize all the way down to AsBlock/IpBlock?

Reply:
I like these to be readable (well a json value of readable) for debugging and use in configuration files.